### ISSUE TITLE
No go on GNOME 41

### ISSUE THREAD

**Issue by stejoo** [tomflannaghan/proxy-switcher#23]
Hi Tom,

Thanks again for creating this plugin and making the new release.
I have been having issues with your proxy-switcher for a few months now. Tried the latest committed to the  `master` branch when it first broke for me (December/January I believe). Probably because of some update. No dice. Just now I tried your new release, hopeful it would resolve my issue but still I can't seem to get it activated in my GNOME. I am currently running Fedora 35 with GNOME 41.5. 

I usually install it manually (well.. automated via Ansible Playbook). Decided to try it by hand by downloading the `.zip` and installing it using the `gnome-extensions` utility that ships with GNOME nowadays. I called  `$ gnome-extensions install ProxySwitcher@flannaghan.com.zip` but still no Proxy Switcher in my top right menu. I also tried the `enable` option of the `gnome-extensions` utility but that didn't accomplish anything either. All the while looking at the logs via `sudo journalctl -f /usr/bin/gnome-shell`. However the log remains silent on the plugin front... perhaps I need to up a log level somewhere? Going fully manual by extracting the plugin myself in `~/.local/share/gnome-shell/extensions` also didn't work.

I uninstalled and cleaned up. And went to try the "should just work" method: go to https://extensions.gnome.org/extension/771/proxy-switcher/ and flicked the Off/On button to the On position. It asked for my permission to install the plugin, the switch flicked to On but... nothing still. Refresh of the page turns to switch to the Off position once more. Hmmm.

So I would like to try and debug it. Hoping you could provide me some pointers on how to debug the issue. I assume you have some way of debugging the plugin during development. 

To sum up (TL;DR):

1. The latest release does not work for me. Not the one published here, not the one on the GNOME Extension website by flicking the switch. System is Fedora 35 with GNOME 41.5. No other extensions installed at all, not any themes, my system should be pretty close to stock.
2. Do you know any way to debug the issue? I simply have no experience with GNOME Shell extension debugging. If you could give me some pointers those would be quite welcome. Perhaps I can have a go at figuring out what goes awry here.
3. Any other tips to get it to work? I am all ears.

And one last question: can I buy you a beer later? (or donate one :wink: )

**Comment by tomflannaghan** [comment]
Hi @stejoo,

Sorry to hear you've had problems with it. I also run Fedora 35 and it works fine on my system (and on a VM I use to check against a completely stock system) so not sure what it could be.

The way to debug Gnome Extensions is to hit Alt-F2, and enter the command `lg`. This brings up the debug window for Gnome. There's an "Extensions" tab which should show ProxySwitcher and any errors. If you could post any errors shown there that would be really useful (you have to click "Show Errors" to see them).

The extension does assume some things about the existing contents of the dropdown panel it appears on (in order to get it into a sensible location within the panel) so may be something to do with that? If you could take a screenshot of your panel that might also help.

Thanks,
Tom

**Comment by stejoo** [comment]
Tom, thanks for swift reply and the tips. I will remember `lg`the next time I run into something strange with GNOME.

I just spoke to @rduivenvoorde (from issue #19) and he mentioned there is a "grand toggle" for the GNOME extensions.
That switch did not show up in my GNOME nor the GNOME Extensions website, and I do not have the "Extensions" GUI he has on Debian. But upon him showing me that grand switch I realized this must be some GSetting that en-/disables all extensions.

I figured out how to toggle it on my box:
```
$ gsettings set org.gnome.shell disable-user-extensions false
```

That enabled all extensions pre-installed by Fedora and your proxy-switcher popped up just fine.

One question remains: can I buy you a beer or donate you one?

**Comment by tomflannaghan** [comment]
That also happened to me and was very confusing at the time - a very subtle change of behaviour. I bet it caught a lot of people out.

If you want you can install the Extensions GUI with the global toggle on Fedora, although your gsettings command does the job perfectly well. It's a seperate package - gnome-extensions-app.

No worries about the beer :-) I'm glad you enjoy the extension!
